File: web2py/applications/school/src/factory/Factory.ts
import RegisterStudentFormModel from '../models/RegisterStudentFormModel'

class Factory {
  static createRegisterStudentForm (): RegisterStudentFormModel {
    return new RegisterStudentFormModel(
      { name: 'name', type: 'text', label: 'Name', value: "", id: "name"},
      { name: 'grade', type: 'text', label: 'Grade', value: "", id: "grade"},
      { name: 'age', type: 'text', label: 'Age', value: "", id: "age"})
  }
}

export default Factory

File: web2py/applications/school/modules/factory/Factory.py
from models.Models import Assistance, Student


class Factory:
  @staticmethod
  def create_student(name, grade, age):
    return Student(name=name, grade=grade, age=age)

  @staticmethod
  def create_assistance(student_id, subject_id, classroom_id, week_number, day_of_week, assists):
    return Assistance(
      student_id=student_id,
      subject_id=subject_id,
      classroom_id=classroom_id,
      week_number=week_number,
      day_of_week=day_of_week,
      assists=assists,
    )
